gcc/d/d-objfile.cc

```diff
@@ -796,27 +810,20 @@ class DeclVisitor : public Visitor
 	return;
       }
 
-    /* This function exists in static storage.
-       (This does not mean `static' in the C sense!)  */
-    TREE_STATIC (fndecl) = 1;
-
-    /* Add this decl to the current binding level.  */
-    d_pushdecl (fndecl);
-
     /* Start generating code for this function.  */
     gcc_assert (d->semanticRun == PASSsemantic3done);
     d->semanticRun = PASSobj;
 
-    /* Nested functions may not have its body compiled before the outer
-       function is finished.  GCC requires that nested functions be finished
-       first so we need to arrange for build_decl_tree to be called earlier.  */
-    FuncDeclaration *fdp = d->toParent2 ()->isFuncDeclaration ();
-    if (fdp && fdp->semanticRun < PASSobj)
-      fdp->accept (this);
-
     if (global.params.verbose)
```

Comment:
Hurr, that was a fix-up to #436.  It's never really been necessary to do this, I'll split it out to a separate PR.